pkg/auth/handler/webapp/forgot_password.go

```diff
@@ -71,11 +71,12 @@ type ForgotPasswordHandler struct {
 }
 
 func (h *ForgotPasswordHandler) MakeIntent(r *http.Request) *webapp.Intent {
+	redirectURI := r.URL.Query().Get("redirect_uri")
```

Comment:
We should validate whether it's the same origin before using it.

Reply:
It is actually done at the use-site. Only the Path and Query is used. Other url components are ignored.